README.md

```diff
@@ -15,10 +15,70 @@
 ```bash
 $ npm install @adobe/helix-querybuilder
 ```
+## Background
+
+The [AEM QueryBuilder](https://docs.adobe.com/content/help/en/experience-manager-65/developing/platform/query-builder/querybuilder-api.html) is a Java and REST API for executing server-side queries using a custom Query Builder Language (QBL). QBL was designed to be:
+
+1. implementation agnostic
+2. HTML-form friendly (you should not need JavaScript to build a query)
+3. simple (no joins or projections)
+
+## QBL Language and Notation
+
+### As URL Query String
+
+The most common way of expressing queries is as a query string appended to the URL of the resource that is able to execute queries. In the context of Project Helix, this would be the [Helix Data Embed Action](https://github.com/adobe/helix-data-embed#filtering-results).
+
+An example query might look like this:
+
+```
+https://adobeioruntime.net/api/v1/web/helix/helix-services/data-embed@v1/https://blogs.adobe.com/psirt/?feed=atom&hlx_property=author&hlx_property.value=svishnoi
+```
+
+The query is encoded in the URL parameters `hlx_property=author&hlx_property.value=svishnoi`.
+
+### As a multi-line text
+
+This URL query string notation is most practical in day-to-day use, but a bit hard to read. Therefore a multi-line text notation is used that uses line breaks to separate key-value-pairs, does not use prefixes, nor URL-encoding:
+
+```
+property=author
+property.value=svishnoi
+```
+
+### As JSON or YAML
+
+When using QBL in configuration files or JavaScript applications, it can be convenient to represent QBL in JSON like this:
+
+```json
+{
+  "property": {
+    "property": "author",
+    "value": "svishnoi"
+  }
```

Comment:
Might be a dumb question, but why is nesting a `property` in a `property` required? Wouldn't it be more natural to have:
```
"property": {
  "name": "author",
  "value": "svishnoi"
}
```

Reply:
QB is using a shorthand syntax like:

```
property=author
property.value= svishnoi
```

Where `property=` (or `foo=`) is shorthand for `property.property=` (or `foo.foo=`).

The outer `property` is the predicate type.

The goal of the JSON syntax is not to be super short, but to be idiomatic.

Reply:
I would also prefer the `name` variant. I don't think that the `json` representation needs to be the same, but readable.